tracing-opentelemetry-http/src/main/java/io/micronaut/tracing/opentelemetry/instrument/http/client/OpenTelemetryClientFilter.java

```diff
@@ -0,0 +1,131 @@
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import static io.micronaut.tracing.opentelemetry.instrument.http.client.OpenTelemetryClientFilter.CLIENT_PATH;
+
+/**
+ * An HTTP client instrumentation filter that uses Open Telemetry.
+ *
+ * @author Nemanja Mikic
+ */
+@Filter(CLIENT_PATH)
+@Requires(beans = Tracer.class)
+public class OpenTelemetryClientFilter extends AbstractOpenTelemetryFilter implements HttpClientFilter {
+
+    @SuppressWarnings("rawtypes")
+    private final Instrumenter<MutableHttpRequest, HttpResponse> instrumenter;
+
+    /**
+     * Initialize the open tracing client filter with tracer and exclusion configuration.
+     *
+     * @param openTelemetry    the openTelemetry
+     * @param exclusionsConfig The {@link OpenTelemetryExclusionsConfiguration}
+     * @param openTelemetryHttpClientConfig The {@link OpenTelemetryHttpClientConfig}
+     */
+    public OpenTelemetryClientFilter(OpenTelemetry openTelemetry,
+                                     @Nullable OpenTelemetryExclusionsConfiguration exclusionsConfig,
+                                     @Nullable OpenTelemetryHttpClientConfig openTelemetryHttpClientConfig) {
+        super(exclusionsConfig == null ? null : exclusionsConfig.exclusionTest());
+        MicronautHttpClientTelemetryBuilder micronautHttpClientTelemetryBuilder = new MicronautHttpClientTelemetryBuilder(openTelemetry);
+
+        if (openTelemetryHttpClientConfig != null) {
+            micronautHttpClientTelemetryBuilder.setCapturedRequestHeaders(openTelemetryHttpClientConfig.getRequestHeaders());
+            micronautHttpClientTelemetryBuilder.setCapturedResponseHeaders(openTelemetryHttpClientConfig.getResponseHeaders());
+        }
+
+        instrumenter = micronautHttpClientTelemetryBuilder.build();
+    }
+
+    @Override
+    public Publisher<? extends HttpResponse<?>> doFilter(MutableHttpRequest<?> request,
+                                                         ClientFilterChain chain) {
+
+        Publisher<? extends HttpResponse<?>> requestPublisher = chain.proceed(request);
+
+        if (shouldExclude(request.getPath())) {
+            return requestPublisher;
+        }
+
+        Context parentContext = Context.current();
+        if (!instrumenter.shouldStart(parentContext, request)) {
+            return requestPublisher;
+        }
+
+        return (Publishers.MicronautPublisher<HttpResponse<?>>) actual -> {
+            Context context = instrumenter.start(parentContext, request);
+
+            try (Scope ignored = context.makeCurrent()) {
+                requestPublisher.subscribe(new Subscriber<HttpResponse<?>>() {
```

Comment:
should this implement `CoreSubscriber` from Reactor and override `currentContext`?